src/NHibernate/Engine/BatchFetchQueue.cs

```diff
@@ -420,7 +418,8 @@ internal object[] GetEntityBatch(IEntityPersister persister, object id, int batc
 			// List of entity keys that haven't been checked for their existance in the cache. Besides the entity key,
 			// the index where the key was found is also stored in order to correctly order the returning keys.
 			var entityKeys = new List<KeyValuePair<EntityKey, int>>(batchSize);
-			var batchableCache = persister.Cache?.Cache as IBatchableReadOnlyCache;
+			// If there is a cache, obsolete or not, batchableCache will not be null.
+			var batchableCache = persister.Cache?.GetCacheBase();
```

Comment:
In line 238 you use as CacheBase. But here is GetCacheBase? Is it intentional?

Reply:
No, line 238 is an error and causes the last fixup to introduce a potential bug causing cache to be ignored in some cases, thanks for having spotted it.